src/cap32.cpp

```diff
@@ -1568,26 +1568,26 @@ void update_cpc_speed()
 
 std::string getConfigurationFilename(bool forWrite)
 {
-  // First look in any user supplied configuration file path
-  std::string configFilename = args.cfgFilePath;
-  if(access(configFilename.c_str(), F_OK) != 0) {
-     // If not found, cap32.cfg in the same directory as the executable
-     configFilename = std::string(chAppPath) + "/cap32.cfg";
-     // If not found, look for .cap32.cfg in the home of current user
-     if (access(configFilename.c_str(), F_OK) != 0) {
-        configFilename = std::string(getenv("HOME")) + "/.cap32.cfg";
-        // If still not found, look for cap32.cfg in /etc
-        if (!forWrite && access(configFilename.c_str(), F_OK) != 0) {
-           configFilename = "/etc/cap32.cfg";
-        }
-     }
+  int mode = R_OK | ( F_OK * forWrite );
+  std::array<std::string, 6> configFilename = {
+      args.cfgFilePath, // First look in any user supplied configuration file path
+      std::string(chAppPath) + "/cap32.cfg", // If not found, cap32.cfg in the same directory as the executable
+      std::string(getenv("XDG_CONFIG_HOME")) + "/cap32.cfg", // If not found, look for .cap32.cfg in the XDG_CONFIG_HOME of current user  
```

Comment:
I'm wondering what happens when getenv returns NULL.

Reply:
i'm wondering too.

Reply:
I cloned the branch and build it. It turns out it just segfault if any of the variable is not defined:

$ ./cap32
terminate called after throwing an instance of 'std::logic_error'
  what():  basic_string::_M_construct null not valid
Aborted

$ XDG_CONFIG_HOME="." ./cap32
Using configuration file: cap32.cfg
No joystick found.
joysticks_init() failed. Joysticks won't work.

So this needs to be fixed :-)

Reply:
sure.

Reply:
but as you can see, it finds ressources.

Reply:
Yes, because it uses appPath which default to getcwd():
     CPC.resources_path = conf.getStringValue("system", "resources_path", appPath + "/resources");

So this is OK but the debian patch should still be updated. For an install, we do want it to look in /usr/share.